tutorials/arvo/cli-tutorial.md

```diff
@@ -0,0 +1,427 @@
+message means that `~nus` failed to connect to `~zod`'s `%shoe` session.
+
+### `%shoe`'s code
+
+```hoon
+::  shoe: example usage of /lib/shoe
+::
+::    the app supports one command: "demo".
+::    running this command renders some text on all sole clients.
+::
+/+  shoe, verb, dbug, default-agent
+```
+
+`/+` is the Ford rune which imports libraries from the `/lib` directory into
+the subject.
+ * `shoe` is the `shoe` library.
+ * `verb` is a library used to print what a Gall agent is doing.
+ * `dbug` is a library of debugging tools.
+ * `default-agent` contains a Gall agent core with minimal implementations of
+   required Gall arms.
+
+```hoon
+|%
++$  state-0  [%0 ~]
++$  command  ~
+::
++$  card  card:shoe
+--
+```
+The types used by the app.
+
+`state-0` stores the state of the app, which is null as there is no state to
+keep track of. It is good practice to include a version number anyways,
+in case the app is made stateful at a later time.
+
+`command` is typically a set of tagged union types that represent the possible
+commands that can be entered by the user. Since this app only supports one
+command, it is unnecessary for it to have any associated data, thus the command
+is represented by `~`.
+
+In a non-trivial context, a `command` is commonly given by `[%name
+data]`, where `%name` is the identifier for the type of command and `data` is
+a type or list of types that contain data needed to execute the command. See
+`app/chat-cli.hoon` for examples of commands, such as `[%say letter:store]` and
+`[%delete path]`. This is not required though, and you could use something like
+`[chat-room=@t =action]`.
+
+`card` is either an ordinary Gall agent `card` or a `%shoe` `card`, which takes
+the shape `[%shoe sole-ids=(list @ta) effect=shoe-effect]`. A `%shoe` `card` is
+sent to all `sole`s listed in `sole-ids`, imaking them run the `sole-effect`
+specified by `effect` (i.e. printing some text).  Here we can
+reference `card:shoe` because of `/+  shoe` at the beginning of the app.
+
+```hoon
+=|  state-0
+=*  state  -
+::
+```
+Add the bunt value of `state-0` to the head of the subject, then give it the
+macro `state`. The `-` here is a lark expression referring to the head of the
+subject. This allows us to use `state` to refer to the state elsewhere in the
+code no matter what version we're using, while also getting direct access to the
+contents of `state` (if it had any).
+
+```hoon
+%+  verb  |
+%-  agent:dbug
+^-  agent:gall
+%-  (agent:shoe command)
+^-  (shoe:shoe command)
+```
+
+This sequence of commands tells the compiler that the following core is a Gall
+agent core that may possess arms of the sort defined in `verb`, `agent:dbug`,
+and `agent:shoe`. These cores tell the compiler what type signature these arms
+must have, and ensure that the arms that are not standard Gall arms are put into
+the context so that the resulting core nests within `agent:gall`.
```

Comment:
I [still](https://github.com/urbit/docs/pull/912/#discussion_r456804690) feel this paragraph is incorrect and confusing. The paragraph that directly follows is much better. Did you maybe just forget to delete this?

Reply:
Yup, its gone now.